src/memory/combat_manager.rs

```diff
@@ -2,12 +2,95 @@ use crate::memory::memory_context::MemoryContext;
 use crate::memory::{MemoryManager, MemoryManagerUpdate};
 use crate::state::StateContext;
 use log::info;
+use memory::game_engine::il2cpp::unity_list::{UnityItem, UnityList};
 use memory::memory_manager::il2cpp::UnityMemoryManager;
 use memory::process::MemoryError;
+use memory::process::Process;
+use memory::string::*;
+
+#[derive(Default, Debug)]
+pub enum CombatControllerType {
+    #[default]
+    Basic,
+    FirstEncounter,
+    SecondEncounter,
+    DwellerOfStrife,
+    DwellerOfDread,
+    KOTutorial,
+    LiveManaTutorial,
+    ManaRegenTutorial,
+    RoundsTutorial,
+    SpellLockTutorial,
+    TimedBlocksTutorial,
+    TimedHitsTutorial,
+}
+
+#[derive(Debug, Default, Clone)]
+pub enum CombatDamageType {
+    #[default]
+    None = 0,
+    Any = 1,
+    Sword = 2,
+    Sun = 4,
+    Moon = 8,
+    Eclipse = 16,
+    Poison = 32,
+    Arcane = 64,
+    Stun = 128,
+    Blunt = 256,
+    Magical = 252,
+}
+
+impl CombatDamageType {
+    fn from_u32(value: u32) -> CombatDamageType {
+        match value {
+            0 => CombatDamageType::None,
+            1 => CombatDamageType::Any,
+            2 => CombatDamageType::Sword,
+            4 => CombatDamageType::Sun,
+            8 => CombatDamageType::Moon,
+            16 => CombatDamageType::Eclipse,
+            32 => CombatDamageType::Poison,
+            64 => CombatDamageType::Arcane,
+            128 => CombatDamageType::Stun,
+            256 => CombatDamageType::Blunt,
+            252 => CombatDamageType::Magical,
+            _ => CombatDamageType::None,
+        }
+    }
+}
+
+#[derive(Default, Debug)]
+pub struct LiveMana {
+    pub big: u32,
+    pub small: u32,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct CombatEnemy {
+    pub current_hp: u32,
+    pub unique_id: String,
+    pub guid: String,
+    pub max_hp: u32,
+    pub speed: u32,
+    pub physical_attack: u32,
+    pub physical_defense: u32,
+    pub magical_attack: u32,
+    pub magical_defense: u32,
+    pub turns_to_action: u8,
+    pub total_spell_locks: u8,
+    pub spell_locks: UnityList<CombatDamageType>,
+}
```

Comment:
These guys have physical/magical defense stats. Do the enemies have element vulnerabilities too, or is that not a thing? Wyrd is a special case I know.

Should this one not have `is_fleshmancer_minion`? Or is that not present in the "live" data?

Reply:
its present! I'll prepare a secondary PR to get these additional fields added 
![image](https://github.com/user-attachments/assets/a2313d1a-8274-438e-912f-3f0de9f4c708)